docs/core/DelegationManager.md

```diff
@@ -0,0 +1,195 @@
+
+**Requirements**:
+* Caller MUST NOT already be an Operator
+* Caller MUST NOT already be delegated to an Operator
+* `earningsReceiver != address(0)`
+* `stakerOptOutWindowBlocks <= MAX_STAKER_OPT_OUT_WINDOW_BLOCKS`: (~15 days)
+* Pause status MUST NOT be set: `PAUSED_NEW_DELEGATION`
+
+*Unimplemented as of M2*:
+* `require(!slasher.isFrozen(operator))` is currently a no-op
+
+#### `modifyOperatorDetails`
+
+```solidity
+function modifyOperatorDetails(OperatorDetails calldata newOperatorDetails) external
+```
+
+Allows an Operator to update their stored `OperatorDetails`.
+
+**Requirements**:
+* Caller MUST already be an Operator
+* `new earningsReceiver != address(0)`
+* `new stakerOptOutWindowBlocks >= old stakerOptOutWindowBlocks`
+* `new stakerOptOutWindowBlocks <= MAX_STAKER_OPT_OUT_WINDOW_BLOCKS`
+
+#### `updateOperatorMetadataURI`
+
+```solidity
+function updateOperatorMetadataURI(string calldata metadataURI) external
+```
+
+Allows an Operator to emit an `OperatorMetadataURIUpdated` event. No other state changes occur.
+
+**Requirements**:
+* Caller MUST already be an Operator
+
+### Stakers
+
+Stakers interact with the following functions:
+
+#### `delegateTo`
+
+```solidity
+function delegateTo(address operator, SignatureWithExpiry memory approverSignatureAndExpiry, bytes32 approverSalt) external
+```
+
+Allows the caller (a Staker) to delegate ALL their shares to an Operator (delegation is all-or-nothing). For each strategy the Staker has shares in, the `DelegationManager` will update the Operator's corresponding delegated share amounts.
+
+**Effects**:
+* Records the Staker as being delegated to the Operator
+* If the Staker has deposited into the `EigenPodManager` and is not in undelegation limbo, the `DelegationManager` adds these shares to the Operator's shares for the beacon chain ETH strategy.
+* For each of the three strategies in the `StrategyManager`, if the Staker holds shares in that strategy they are added to the Operator's shares under the corresponding strategy.
+
+**Requirements**:
+* The caller MUST NOT already be delegated to an Operator
+* The `operator` MUST already be an Operator
+* If the `operator` has a `delegationApprover`, the caller MUST provide a valid `approverSignatureAndExpiry` and `approverSalt`
+* Pause status MUST NOT be set: `PAUSED_NEW_DELEGATION`
+
+*Unimplemented as of M2*:
+* `require(!slasher.isFrozen(operator))` is currently a no-op
+
+#### `delegateToBySignature`
+
+```solidity
+function delegateToBySignature(
+    address staker,
+    address operator,
+    SignatureWithExpiry memory stakerSignatureAndExpiry,
+    SignatureWithExpiry memory approverSignatureAndExpiry,
+    bytes32 approverSalt
+) external
+```
+
+Allows a Staker to delegate to an Operator by way of signature. This function can be called by three different parties:
+* If the Staker calls this method, they need to submit both the `stakerSignatureAndExpiry` AND `approverSignatureAndExpiry`
```

Comment:
this should be the 'default' case / what happens if called by any address that is neither the operator, nor the operator's delegationApprover (rather than if called specifically by the staker themself)